vector/src/main/java/im/vector/app/features/home/room/list/RoomListViewModel.kt

```diff
@@ -120,40 +120,34 @@ class RoomListViewModel @Inject constructor(
         }
     }
 
-    val sections: List<RoomsSection> by lazy {
-        if (appStateHandler.getCurrentRoomGroupingMethod() is RoomGroupingMethod.BySpace) {
-            SpaceRoomListSectionBuilder(
-                    session,
-                    stringProvider,
-                    appStateHandler,
-                    viewModelScope,
-                    suggestedRoomJoiningState,
-                    autoAcceptInvites,
-                    {
-                        it.disposeOnClear()
-                    },
-                    {
-                        updatableQuery = it
-                    },
-                    vectorPreferences.labsSpacesOnlyOrphansInHome()
-            ).buildSections(initialState.displayMode)
-        } else {
-            GroupRoomListSectionBuilder(
-                    session,
-                    stringProvider,
-                    viewModelScope,
-                    appStateHandler,
-                    autoAcceptInvites,
-                    {
-                        it.disposeOnClear()
-                    },
-                    {
-                        updatableQuery = it
-                    }
-            ).buildSections(initialState.displayMode)
+    private val roomListSectionBuilder = if (appStateHandler.getCurrentRoomGroupingMethod() is RoomGroupingMethod.BySpace) {
```

Comment:
no more `lazy {}` block here, is it OK @BillCarsonFr ?

Reply:
I wanted to avoid setting that up in the viewModel init, I think it's better to avoid doing work in constructor. Can it stay lazy?

Reply:
OK

Reply:
yes